### ISSUE TITLE
"Caterpillar" effect when drawing moving objects.

### ISSUE THREAD

**Issue by VoronFX** [antonpup/Aurora#733]
### What are you experiencing an issue with?

- [x] Aurora
- [ ] Aurora supported game, (Fill in the game)
- [ ] Aurora Wrapper
- [ ] Aurora Profile
- [ ] Aurora supported device, (Fill in problematic devices)
- [ ] Aurora Scripts
- [ ] Aurora Updater
- [ ] Aurora Lighting issue
- [ ] Other, (Fill in another issue here)

### What kind of an issue is it?

- [ ] Crash
- [ ] Performance problems
- [x] Visual bug
- [ ] Localization issue
- [ ] Enhancement suggestion
- [ ] Device issue
- [ ] Other, (Fill in another type of issue here)

I've noticed a 9 copies of the next drawing pattern in different places.
```CSharp
float x_pos = (float)Math.Round((sequence.freeform.X + Effects.grid_baseline_x) * Effects.editor_to_canvas_width);
float y_pos = (float)Math.Round((sequence.freeform.Y + Effects.grid_baseline_y) * Effects.editor_to_canvas_height);
float width = (float)(sequence.freeform.Width * Effects.editor_to_canvas_width);
float height = (float)(sequence.freeform.Height * Effects.editor_to_canvas_height);

Rectangle rect = new Rectangle((int)x_pos, (int)y_pos, (int)width, (int)height);
```

This rounding and use of integer Rectangle was a try to align objects to the edges of pixels as I understand. There are still a lot of unprecise types in drawing, even after this rounding that I am not sure if they are useful. But in what I am sure is that they are causing objects to have different width while moving. It creates effect that I can describe as move of caterpillar. 
Like this:
```
|    |    |    |    |    |    |    |    |    |    |    |

     |==gradient/object==|
     
     |====gradient/object====|
     
         |=gradient/object===|
         
         |====gradient/object====|
         
             |=gradient/object===|
         
             |====gradient/object====|
```

I am not sure which one is better, smoothness of movement or bit more alignment.
What you think about it?

**Comment by simon-wh** [comment]
Are you able to create a demonstration of the possible effects in Aurora? I feel as though the caterpillar effect you describe might not even be noticeable. Personally, smoothness is my number 1, providing the drawbacks are not super noticeable.

**Comment by VoronFX** [comment]
Crap, while creating a demo I've realized that it was my fault. 
Actually missing some of the rounding result in my copy has this effect. 
Also I've just experimented and created the hybrid version that has no caterpillar with width but only inside the object, which makes movement smother by trading the technical accuracy of the gradient.

Anyway at decent moving speed any of those drawing algs are not distinguishable.
I think now issue can be closed, I will leave demo script just if you are curios how it all looks.
[CaterpillarDemo.zip](https://github.com/antonpup/Aurora/files/1220636/CaterpillarDemo.zip)
